test/framework/src/main/java/org/elasticsearch/node/MockNode.java

```diff
@@ -36,26 +41,38 @@
  * </ul>
  */
 public class MockNode extends Node {
-
-    private final boolean mockBigArrays;
-    private Collection<Class<? extends Plugin>> plugins;
+    private final Collection<Class<? extends Plugin>> classpathPlugins;
 
     public MockNode(Settings settings, Collection<Class<? extends Plugin>> classpathPlugins) {
         super(InternalSettingsPreparer.prepareEnvironment(settings, null), classpathPlugins);
-        this.plugins = classpathPlugins;
-        this.mockBigArrays = classpathPlugins.contains(NodeMocksPlugin.class); // if this plugin is present we mock bigarrays :)
```

Comment:
This didn't work because we use the value of `mockBigArrays` in the `super` call above. But we hadn't yet set `mockBigArrays`.